python/ql/lib/semmle/python/dataflow/new/FlowSummary.qll

```diff
@@ -67,32 +69,44 @@ abstract class SummarizedCallable extends LibraryCallable, Impl::Public::Summari
 }
 
 class RequiredSummaryComponentStack = Impl::Public::RequiredSummaryComponentStack;
-//
-// TODO: Implement this
-//
-// private class SummarizedCallableFromModel extends SummarizedCallable {
-//   string package;
-//   string type;
-//   string path;
-//   SummarizedCallableFromModel() {
-//     ModelOutput::relevantSummaryModel(package, type, path, _, _, _) and
-//     this = package + ";" + type + ";" + path
-//   }
-//   override Call getACall() {
-//     exists(API::MethodAccessNode base |
-//       ModelOutput::resolvedSummaryBase(package, type, path, base) and
-//       result = base.getCallNode().asExpr().getExpr()
-//     )
-//   }
-//   override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
-//     exists(string kind |
-//       ModelOutput::relevantSummaryModel(package, type, path, input, output, kind)
-//     |
-//       kind = "value" and
-//       preservesValue = true
-//       or
-//       kind = "taint" and
-//       preservesValue = false
-//     )
-//   }
-// }
+
+// This gives access to getNodeFromPath, which is not constrained to `CallNode`s
+// as `resolvedSummaryBase` is.
+private import semmle.python.frameworks.data.internal.ApiGraphModels as AGM
```

Comment:
as you say in the commit message, no tests :disappointed: Could we just leave this code commented out until we have time/energy to show that it works as intended? (I might also be persuaded with a promise that you make a followup PR with tests...)

Reply:
I think leaving it (commented) out is probably the right thing to do. Then a follow-up PR can properly demonstrate what it enables.